src/components/ha-camera-stream.ts

```diff
@@ -0,0 +1,153 @@
+import { property, UpdatingElement, PropertyValues } from "lit-element";
+
+import computeStateName from "../common/entity/compute_state_name";
+import { HomeAssistant, CameraEntity } from "../types";
+import { fireEvent } from "../common/dom/fire_event";
+import { fetchStreamUrl, computeMJPEGStreamUrl } from "../data/camera";
+
+type HLSModule = typeof import("hls.js");
+
+class HaCameraStream extends UpdatingElement {
+  @property() public hass?: HomeAssistant;
+  @property() public stateObj?: CameraEntity;
+  @property() public showControls: boolean = false;
+  private _hlsPolyfillInstance?: Hls;
+
+  public connectedCallback() {
+    super.connectedCallback();
+    // @ts-ignore
+    if (!this._hasRequestedUpdate) {
+      this.requestUpdate();
+    }
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._teardownPlayback();
+  }
+
+  protected update(changedProps: PropertyValues) {
+    super.update(changedProps);
+    if (this.lastChild) {
```

Comment:
This means that if I would set `stateObj` to `undefined`, you won't tear down the player.

Reply:
The teardown is happening in the disconnected callback, which happens for `more-info` as well.  Can add the check back, but this is really the only way I could get everything to work correctly.  I'm not sure what else I can do here.  Any hints?

Reply:
Don't build components with a specific use case in mind.

We should be able to have a `<ha-camera-stream>` in the DOM and set it's stateObj to undefined and not detach it from the DOM, but maybe hide it with CSS, and things should still be able to work.